### ISSUE TITLE
no-unpublished-require/no-unpublished-bin false positives

### ISSUE THREAD

**Issue by aarongoldenthal** [eslint-community/eslint-plugin-n#122]
In published `bin` files in npm packages, there are false positive for the `no-unpublished-require` and `no-unpublished-bin` rules when a path is included in the package.json `files` array relative to `.`.

For example, with package.json:

```json
  "bin": "./bin/foo.js",
  "exports": "./index.js",
  "files": [
    "index.js",
    "bin/",
    "lib/"
  ]
```

The following `./bin/foo.js` does not show any `no-unpublished-require` errors:

```js
#!/usr/bin/env node
'use strict';

const foo = require('..');
const bar= require('../lib/bar.js');
```

If the package.json is changed to:

```json
  "bin": "./bin/foo.js",
  "exports": "./index.js",
  "files": [
    "./index.js",
    "bin/",
    "./lib/"
  ]
```

then `./bin/foo.js` shows a `no-unpublished-require` error for both `require`s. This seems to only occur in `bin` files. For example, if `./lib/bar.js` includes a `require('../') it doe snot show a `no-unpublished-require` error. The required files are, however, published.

Similarly, if the `files` entry `bin/` is changed to `./bin/`, then the entire `./bin/foo.js` file shows a `no-unpublished-bin` error (it seems related).

There's an example at https://github.com/aarongoldenthal/eslint-n-require-false-positive.

**Comment by scagood** [comment]
This looks to be the same as #99

I will see if I can find some time this week to look
